lib/jwt/token/verifier.rb

```diff
@@ -10,7 +10,10 @@ def self.included(base)
 
       module ClassMethods
         def verify(jwt_token, context = nil)
-          decoded = JWT.decode(jwt_token, configuration.secret[:public], true, decode_options)
+          decoded = JWT.decode(jwt_token, nil, true, decode_options) do |header|
+            algorithm_type = JWT::Token::Configuration::ALGORITHMS[header["alg"]]
+            configuration.send(algorithm_type).public_key if algorithm_type
```

Comment:
If `algorithm_type ` is absent we get in line 18 `NoMethodError (undefined method '[]' for nil:NilClass)`. I would rather raise error here that algorithm type is absent.

Reply:
JWT will actually return an error: https://github.com/jwt/ruby-jwt/blob/master/lib/jwt/decode.rb#L63